src/System.Diagnostics.TextWriterTraceListener/src/System/Diagnostics/XmlWriterTraceListener.cs

```diff
@@ -0,0 +1,375 @@
+﻿using System.Collections;
+using System.Globalization;
+using System.IO;
+using System.Runtime.Versioning;
+using System.Text;
+using System.Threading;
+using System.Xml;
+using System.Xml.XPath;
+
+namespace System.Diagnostics
+{
+    public class XmlWriterTraceListener : TextWriterTraceListener
+    {
+        private const string FixedHeader = "<E2ETraceEvent xmlns=\"http://schemas.microsoft.com/2004/06/E2ETraceEvent\"><System xmlns=\"http://schemas.microsoft.com/2004/06/windows/eventlog/system\">";
+
+        private readonly string _machineName = Environment.MachineName;
+        private StringBuilder _strBldr = null;
+        private XmlTextWriter _xmlBlobWriter = null;
+
+        public XmlWriterTraceListener(Stream stream)
+            : base(stream)
+        {
+        }
+
+        public XmlWriterTraceListener(Stream stream, string name)
+            : base(stream, name)
+        {
+        }
+
+        public XmlWriterTraceListener(TextWriter writer)
+            : base(writer)
+        {
+        }
+
+        public XmlWriterTraceListener(TextWriter writer, string name)
+            : base(writer, name)
+        {
+        }
+
+        [ResourceExposure(ResourceScope.Machine)]
+        [ResourceConsumption(ResourceScope.Machine)]
+        public XmlWriterTraceListener(string filename)
+            : base(filename)
+        {
+        }
+
+        [ResourceExposure(ResourceScope.Machine)]
+        [ResourceConsumption(ResourceScope.Machine)]
+        public XmlWriterTraceListener(string filename, string name)
+            : base(filename, name)
+        {
+        }
+
+        public override void Write(string message)
+        {
+            WriteLine(message);
+        }
+
+        public override void WriteLine(string message)
+        {
+            TraceEvent(null, SR.TraceAsTraceSource, TraceEventType.Information, 0, message);
+        }
+
+        public override void Fail(string message, string detailMessage)
+        {
+            int length = detailMessage != null ? message.Length + 1 + detailMessage.Length : message.Length;
+            TraceEvent(null, SR.TraceAsTraceSource, TraceEventType.Error, 0, string.Create(length, detailMessage,
+            (dst, v) =>
+            {
+                ReadOnlySpan<char> prefix = message;
+                prefix.CopyTo(dst);
+
+                if (v != null)
+                {
+                    ReadOnlySpan<char> space = " ";
+                    space.CopyTo(dst.Slice(prefix.Length, 1));
```

Comment:
Seems like this would be much simpler as just:
```C#
dst[prefix.Length] = ' ';
```

Reply:
Right. Thanks